PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/component/table/AppSettingModel.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) Microsoft Corporation
+ * <p/>
+ * All rights reserved.
+ * <p/>
+ * MIT License
+ * <p/>
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * <p/>
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ * <p/>
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.intellij.runner.functions.component.table;
+
+
+import org.apache.commons.collections4.ListUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+
+import javax.swing.event.TableModelListener;
+import javax.swing.table.TableModel;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AppSettingModel implements TableModel {
+
+    private static final String[] TITLE = {"Key", "Value"};
+
+    private List<Pair<String, String>> appSettings = new ArrayList<>();
+
+    @Override
+    public int getRowCount() {
+        return appSettings.size();
+    }
+
+    @Override
+    public int getColumnCount() {
+        return 2;
+    }
+
+    @Override
+    public String getColumnName(int col) {
+        return TITLE[col];
+    }
+
+    @Override
+    public Class<?> getColumnClass(int col) {
+        return String.class;
+    }
+
+    @Override
+    public boolean isCellEditable(int row, int col) {
+        return true;
+    }
+
+    @Override
+    public Object getValueAt(int row, int col) {
+        final Pair<String, String> target = appSettings.get(row);
+        return col == 0 ? target.getKey() : target.getValue();
+    }
+
+    @Override
+    public void setValueAt(Object value, int row, int col) {
+        while (row >= appSettings.size()) {
```

Comment:
throw IAE error if value is not null and is not a string

Reply:
Added